File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/EditMealInteractor.java
package ardjomand.leonardo.nutrimeal.data;

import android.net.Uri;

import ardjomand.leonardo.nutrimeal.data.pojos.Meal;

public interface EditMealInteractor {

    interface Interactor {

        void updateMeal(Meal meal);

        void updateMealImage(String key, Uri imageUri);

        void deleteMeal(String key);
    }

    interface Presenter {

        void onItemRead(Meal meal);
    }

}

File: app/src/main/java/ardjomand/leonardo/nutrimeal/data/EditMealInteractor.kt
package ardjomand.leonardo.nutrimeal.data

import android.net.Uri

import ardjomand.leonardo.nutrimeal.data.pojos.Meal

interface EditMealInteractor {

    interface Interactor {

        fun updateMeal(meal: Meal)

        fun updateMealImage(key: String, imageUri: Uri)

        fun deleteMeal(key: String)
    }

    interface Presenter {

        fun onItemRead(meal: Meal)
    }

}
